vmdb/app/models/mixins/has_many_orchestration_stack_mixin.rb

```diff
@@ -0,0 +1,15 @@
+module HasManyOrchestrationStackMixin
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :orchestration_stacks,
+             :foreign_key => :ems_id,
+             :dependent   => :destroy
+
+    has_many :direct_orchestration_stacks,
+             :foreign_key => :ems_id,
+             :conditions  => OrchestrationStack.arel_table[:ancestry].eq(nil).to_sql,
+             :dependent   => :destroy,
```

Comment:
the dependent destroy is probably needed only for :orchestration_stacks, right? As this relation is subset of that.